src/services/completions.ts

```diff
@@ -1688,7 +1688,16 @@ namespace ts.Completions {
 
             if (contextToken.kind === SyntaxKind.GreaterThanToken && contextToken.parent) {
                 if (contextToken.parent.kind === SyntaxKind.JsxOpeningElement) {
-                    return true;
+                    // Two possibilities:
+                    //   1. <div>/**/
+                    //      - contextToken: GreaterThanToken (before cursor)
+                    //      - location: JSXElement
+                    //      - different parents (JSXOpeningElement, JSXElement)
+                    //   2. <Component<string> /**/>
+                    //      - contextToken: GreaterThanToken (before cursor)
+                    //      - location: GreaterThanToken (after cursor)
+                    //      - same parent (JSXOpeningElement)
+                    return location.parent !== contextToken.parent;

```

Comment:
Personally, I find parent equality a confusing way to think about this.  Would it be clearer to check whether `location.parent.kind !== SyntaxKind.JsxOpeningElement`?

Reply:
I think both are confusing, but I’m not attached to my way. What I really wanted to do was see if `contextToken` was contained in `contextToken.parent.typeArguments`, but `typeArguments` doesn’t contain the angle brackets at all.

Reply:
Hmm, I thought a nicer comment would be possible for the kind check, but I've utterly failed to come up with one.  Agreed that neither is obviously clearer than the other.